File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
from collections import defaultdict

def solution(survey, choices):
    answer = ''
    dic = defaultdict(int)
    for i in range(len(survey)):
        type, choice = survey[i], choices[i]
        if choice < 4:
            dic[type[0]] += 4 - choice
        elif choice > 4:
            dic[type[1]] += choice - 4
    for type in ["RT", "CF", "JM", "AN"]:
        if dic[type[1]] > dic[type[0]]: answer += type[1]
        else: answer += type[0]
    return answer

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        Map<String, Integer> map = new HashMap<>();
        for (int i=0;i<survey.length;i++) {
            String[] temp = survey[i].split("");
            int choice = choices[i];
            
            if (choice > 4) {
                map.put(temp[1], map.getOrDefault(temp[1], 0)+choice-4);
            } else if (choice < 4) {
                map.put(temp[0], map.getOrDefault(temp[0], 0)+4-choice);
            }
        }
        
        String answer = "";
        String[] types = {"RT", "CF", "JM", "AN"};
        
        for (String type: types) {
            String[] temp = type.split("");
            if (map.getOrDefault(temp[0], 0) < map.getOrDefault(temp[1], 0)) {
                answer += temp[1];
            } else{
                answer += temp[0];
            }
        }
        return answer;
    }
}